ludwig/encoders/image_encoders.py

```diff
@@ -40,84 +39,89 @@ def register(cls, name):
-            conv_layers=None,
-            num_conv_layers=None,
-            filter_size=3,
-            num_filters=32,
-            strides=(1, 1),
-            padding='valid',
-            dilation_rate=(1, 1),
-            conv_use_bias=True,
-            conv_weights_initializer='glorot_uniform',
-            conv_bias_initializer='zeros',
-            conv_weights_regularizer=None,
-            conv_bias_regularizer=None,
-            conv_activity_regularizer=None,
-            # conv_weights_constraint=None,
-            # conv_bias_constraint=None,
-            conv_norm=None,
-            conv_norm_params=None,
-            conv_activation='relu',
-            conv_dropout=0,
-            pool_function='max',
-            pool_size=(2, 2),
-            pool_strides=None,
-            fc_layers=None,
-            num_fc_layers=1,
-            fc_size=128,
-            fc_use_bias=True,
-            fc_weights_initializer='glorot_uniform',
-            fc_bias_initializer='zeros',
+            img_height: int,
+            img_width: int,
+            conv_layers: Optional[List[Dict]] = None,
+            num_conv_layers: Optional[int] = None,
+            first_in_channels: int = None,
+            out_channels: int = 32,
+            kernel_size: Union[int, Tuple[int]] = 3,
+            stride: Union[int, Tuple[int]] = 1,
+            padding: Union[int, Tuple[int], str] = 'valid',
+            dilation: Union[int, Tuple[int]] = 1,
+            conv_bias: bool = True,
+            padding_mode: str ='zeros',
+            conv_norm: Optional[str] = None,
+            conv_norm_params: Optional[Dict[str, Any]] = None,
+            conv_activation: str = 'relu',
+            conv_dropout: int = 0,
+            pool_function: str = 'max',
+            pool_kernel_size: Union[int, Tuple[int]] = 2,
+            pool_stride: Union[int, Tuple[int]] = None,
+            pool_padding: Union[int, Tuple[int], str]  ='valid',
+            pool_dilation: Union[int, Tuple[int]] = 1,
+            groups: int = 1,
+            fc_layers: Optional[List[Dict]] = None,
+            num_fc_layers: Optional[int] = 1,
+            fc_size: int = 128,
+            fc_use_bias: bool = True,
+            fc_weights_initializer: str = 'glorot_uniform',
+            fc_bias_initializer: str = 'zeros',
             fc_weights_regularizer=None,
             fc_bias_regularizer=None,
             fc_activity_regularizer=None,
-            # fc_weights_constraint=None,
-            # fc_bias_constraint=None,
-            fc_norm=None,
-            fc_norm_params=None,
-            fc_activation='relu',
-            fc_dropout=0,
+            fc_norm: Optional[str] = None,
+            fc_norm_params: Optional[Dict[str, Any]] = None,
+            fc_activation: str = 'relu',
+            fc_dropout: float = 0,
             **kwargs
     ):
         super().__init__()
 
         logger.debug(' {}'.format(self.name))
 
+        assert first_in_channels is not None
```

Comment:
Generally prefer not to use assertions because:

- They cannot be recovered from using try-except
- They can be disabled at runtime based on user config

If this is something we want to check, and not just a sanity check for us as developers, then we should make this a `ValueError`.

Reply:
Sounds good -- fixed in the latest commit.